root/edit/EditIndex.js

```diff
@@ -0,0 +1,182 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import Layout from '../layout';
+import FormSubmit from '../components/FormSubmit';
+import RequestLogin from '../components/RequestLogin';
+import linkedEntities from '../static/scripts/common/linkedEntities';
+import EditLink from '../static/scripts/common/components/EditLink';
+import EditorLink from '../static/scripts/common/components/EditorLink';
+import getVoteName from '../static/scripts/edit/utility/getVoteName';
+import {editorMayAddNote, editorMayVote}
+  from '../utility/edit';
+import formatUserDate from '../utility/formatUserDate';
+
+import EditHeader from './components/EditHeader';
+import EditNotes from './components/EditNotes';
+import EditSidebar from './components/EditSidebar';
+import Vote from './components/Vote';
+import VoteTally from './components/VoteTally';
+import chooseEditDetailsComponent from './utility/chooseEditDetailsComponent';
+
+type Props = {
+  +$c: CatalystContextT,
+  +edit: {...EditT, +id: number},
+  +fullWidth?: boolean,
+  +isSummary?: boolean,
+  +voter?: UnsanitizedEditorT,
+};
+
+const EditIndex = ({
+  $c,
+  edit,
+  fullWidth = false,
+}: Props): React.Element<typeof Layout> => {
+  const canAddNote = Boolean($c.user && editorMayAddNote(edit, $c.user));
+  const isOwnEdit = Boolean($c.user && $c.user.id === edit.editor_id);
+  const canVote = Boolean($c.user && editorMayVote(edit, $c.user));
+  const DetailsComponent = chooseEditDetailsComponent(edit);
+
+  return (
+    <Layout fullWidth={fullWidth} title={texp.l('Edit #{id}', {id: edit.id})}>
+      <div id="content">
+        <EditHeader $c={$c} edit={edit} />
+
+        <h2>{l('Changes')}</h2>
+        {edit.data ? DetailsComponent : (
+          <>
+            <p>{l('An error occurred while loading this edit.')}</p>
```

Comment:
I added `die '';` to `sub restore` in an edit class and got this message to show, so that works at least.